File: JCON/KotlinMigration/src/main/java/nl/blue4it/basic/service/async/RestClient.java
package nl.blue4it.basic.service.async;

import example.avro.Payment;
import nl.blue4it.basic.service.exception.SomeOtherClientException;
import org.springframework.stereotype.Component;

@Component
public class RestClient {

    public void send(Payment payment) {
        System.out.println("Rest client started");
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            System.out.println("InterruptedException");
            throw new RuntimeException(e);
        }
        System.out.println("Rest client executed");
    }
}

File: JCON/KotlinMigration/src/main/java/nl/blue4it/basic/service/async/RestClient.kt
package nl.blue4it.basic.service.async

import example.avro.Payment
import kotlinx.coroutines.delay
import nl.blue4it.basic.service.exception.SomeOtherClientException
import org.springframework.stereotype.Component

@Component
class RestClient {
    suspend fun send(payment: Payment?) {
        println("Payment REST")
        delay(1000)
        println("Payment REST executed")
    }
}